bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/internal/ResolHandlerFactory.java

```diff
@@ -0,0 +1,97 @@
+import org.openhab.binding.resol.handler.ResolThingHandler;
+import org.openhab.binding.resol.internal.discovery.ResolDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ResolHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.resol", service = ThingHandlerFactory.class)
+public class ResolHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(ResolHandlerFactory.class);
+
+    private @Nullable LocaleProvider localeProvider;
+
+    @Reference
+    protected void setLocaleProvider(final LocaleProvider localeProvider) {
+        this.localeProvider = localeProvider;
+    }
+
+    protected void unsetLocaleProvider(final LocaleProvider localeProvider) {
+        this.localeProvider = null;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(ResolBindingConstants.THING_TYPE_UID_DEVICE)) {
+            return new ResolThingHandler(thing);
+        }
+
+        if (thingTypeUID.equals(ResolBindingConstants.THING_TYPE_UID_EMU_EM)) {
+            return new ResolEmuEMThingHandler(thing);
+        }
+
+        if (thingTypeUID.equals(ResolBindingConstants.THING_TYPE_UID_BRIDGE)) {
+            ResolBridgeHandler handler = new ResolBridgeHandler((Bridge) thing, localeProvider);
+            registerThingDiscovery(handler);
+            return handler;
+        }
+
+        return null;
+    }
+
+    private synchronized void registerThingDiscovery(ResolBridgeHandler bridgeHandler) {
+        ResolDiscoveryService discoveryService = new ResolDiscoveryService(bridgeHandler);
+        logger.trace("Try to register VBUS Discovery service on BundleID: {} Service: {}",
+                bundleContext.getBundle().getBundleId(), DiscoveryService.class.getName());
+
+        Hashtable<@Nullable String, String> prop = new Hashtable<@Nullable String, String>();
+
+        bundleContext.registerService(DiscoveryService.class.getName(), discoveryService, prop);
+        discoveryService.activate();
+    }
```

Comment:
Can you implement registering the discovery bound to a bridge as a `ThingHandlerService`. See this preliminary documentation on how to do that: https://deploy-preview-1262--openhab-docs-preview.netlify.app/docs/developer/bindings/#discovery-that-is-bound-to-a-thing

Reply:
ok, I tried that, can you please check it? What I recognized, was that bridge defined in a thing file was still detected (even though I changed the representationproperty to the ipAddress) but when I added the discovered one it magically disappeared... So maybe I didn't do it right fully.